token/token.go

```diff
@@ -538,16 +539,16 @@ type NumberValue struct {
 	Text  string
 }
 
-func ToNumber(value string) *NumberValue {
+func ToNumber(value string) (*NumberValue, error) {
```

Comment:
This change looks like a hammer to fix a scratch.

You are updating the signature of an exported method, it would have consequence.

I would have checked the ErrRange error in this method, and set a boolean variable NeedQuote on the NumberValue struct, then use it in IsQuoteNeeded

Reply:
Thank you for your commnet, @ccoVeille! Hmm, you are right. I haven't thought about the possibility that people use the method, but let me think about it a little more carefully 👍